web/src/components/tasks/anytime-action/tax-clearance-certificate/TaxClearanceSteps.tsx

```diff
@@ -1,45 +1,73 @@
 import { HorizontalStepper } from "@/components/njwds-extended/HorizontalStepper";
+import { AnytimeActionTaxClearanceCertificateAlert } from "@/components/tasks/anytime-action/tax-clearance-certificate/AnytimeActionTaxClearanceCertificateAlert";
+import { getAllFieldsNonEmpty } from "@/components/tasks/anytime-action/tax-clearance-certificate/helpers";
 import { CheckEligibility } from "@/components/tasks/anytime-action/tax-clearance-certificate/steps/CheckEligibility";
 import { Download } from "@/components/tasks/anytime-action/tax-clearance-certificate/steps/Download";
 import { Requirements } from "@/components/tasks/anytime-action/tax-clearance-certificate/steps/Requirements";
 import { Review } from "@/components/tasks/anytime-action/tax-clearance-certificate/steps/Review";
+import { useConfig } from "@/lib/data-hooks/useConfig";
 import { StepperStep } from "@/lib/types/types";
-import { ReactElement, useState } from "react";
+import { TaxClearanceCertificateData } from "@businessnjgovnavigator/shared/taxClearanceCertificate";
+import { FormEvent, ReactElement, useState } from "react";
 
 interface Props {
-  steps: StepperStep[];
+  taxClearanceCertificateData: TaxClearanceCertificateData;
   certificatePdfBlob?: Blob;
-  currentStep: number;
-  stepIndex: (value: ((prevState: number) => number) | number) => void;
   saveTaxClearanceCertificateData: () => void;
-  setStepIndex: (step: number) => void;
+  isValid: () => boolean;
+  getInvalidFieldIds: () => string[];
+  onSubmit: (event?: FormEvent<HTMLFormElement>) => void;
+  CMS_ONLY_stepIndex?: number;
 }
 
 export const TaxClearanceSteps = (props: Props): ReactElement => {
+  const { Config } = useConfig();
+
+  const [stepIndex, setStepIndex] = useState(props.CMS_ONLY_stepIndex ?? 0);
   const [certificatePdfBlob, setCertificatePdfBlob] = useState<Blob | undefined>(
     props.certificatePdfBlob || undefined,
   );
 
   const onStepClick = (step: number): void => {
-    if (step === 2 && props.currentStep === 1) {
-      props.saveTaxClearanceCertificateData();
+    props.saveTaxClearanceCertificateData();
+    setStepIndex(step);
+  };
+
+  const onSave = (): void => {
+    props.saveTaxClearanceCertificateData();
+    if (props.isValid()) {
+      setStepIndex(2);
```

Comment:
This line of code only gets invoked when the if condition is true, which is causing some odd behavior: I’m unable to use the primary button to move forward to the next step, but I can still navigate forward using the stepper when an error exist on the eligibility tab.

Reply:
I opted to kill this. I don't think anyone's going to miss it.